app/View/Components/PaginationLimitSelector.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace KBox\View\Components;
+
+use Illuminate\View\Component;
+use Illuminate\Support\Facades\Route;
+
+class PaginationLimitSelector  extends Component
+{
+    /**
+     * Create a new component instance.
+     *
+     * @return void
+     */
+    
+    public $routeName;
+    
+    public $routeParamId;
+
+    public $search_replica_parameters;
+
+    public $pageParams;
+    
+    public $optionItemsPerPage;
+
+    public function __construct($pageParams = [])
+    {
+        $this->routeName = $this->getRouteName();
+
+        $this->optionItemsPerPage = auth()->user()->optionItemsPerPage();
+        
+        $this->pageParams =array_merge(
+
+            $pageParams,
+            [
+                'group' => $this->getParamId(),
+                'search_replica_parameters' => request()->only('s')
```

Comment:
> @najiba-f I went deeply through the code and I did some manual test. There are some minor change to apply in order to prevent regressions on some routes. One comment is on the UI side. It is not blocking, but would be good if you can apply it.
> 
> Considering the order of elements added to url of each button a test code must be changed, the test in question is [`RecentDocumentsTest@test_recent_support_search_parameters`](https://github.com/k-box/k-box/blob/0a3a8e9642d0c1a628de63b928c0297da6a83b1b/tests/Feature/RecentDocumentsTest.php#L289). The test ensures that the links for the number of items selector has the correct values. Based on the changes you did I think the order of the parameters will be `range`, `s` and then `n`, so the lines [L311-L313](https://github.com/k-box/k-box/blob/0a3a8e9642d0c1a628de63b928c0297da6a83b1b/tests/Feature/RecentDocumentsTest.php#L311-L313) should be changed, probably as in the next code block
> 
> ```
> $response->assertSee(route('documents.recent', ['range' => 'currentweek', 's' => 'hello', 'n' => 12]));
>         $response->assertSee(route('documents.recent', ['range' => 'currentweek', 's' => 'hello', 'n' => 24]));
>         $response->assertSee(route('documents.recent', ['range' => 'currentweek', 's' => 'hello', 'n' => 50]));
> ```

@avvertix  I hade made some other chages to the [`RecentDocumentsTest@test_recent_support_search_parameters`](https://github.com/k-box/k-box/blob/0a3a8e9642d0c1a628de63b928c0297da6a83b1b/tests/Feature/RecentDocumentsTest.php#L289). Because, I had to put the 'visibility' => 'public' to the $pageParams array.
```
 $response->assertSee(route('documents.recent', ['visibility'=>'public','range' => 'currentweek', 's' => 'hello', 'n' => 12]));
        $response->assertSee(route('documents.recent', ['visibility'=>'public','range' => 'currentweek', 's' => 'hello', 'n' => 24]));
        $response->assertSee(route('documents.recent', ['visibility'=>'public','range' => 'currentweek', 's' => 'hello', 'n' => 50]));
```

Reply:
The visibility parameter controls where the search is made:

- `private` means on the K-Box
- `public` means on the K-Link

Following that the visibility must not be added to the recent route, especially the `public` one. The recent route should not have a visibility parameter, but I might allow it if the value is `private`.

Reply:
The test was failing because of missing the parameter 'visibility',  that's why I have put it there. And also, on document-layout.blade.php where I am passing  parameter `range`   
```
<x-pagination-limit-selector  :page-params="isset($range) && $range ? ['range'=> $range] : []"/>
```
There are variable `$current_visibility` in document-layout.blade.php which I think handles visibility, should I pass it the same way as `$range` instead putting it in `PaginatorLimitSelector.php`? I think this will reduce the test failing for   `RecentDocumentsTest.php`. What do you think?

Reply:
hmm better to use the visibility value of `$current_visibility` if available and not insert anything if the `$current_visibility` is undefined

Reply:
Okay, will do.